master/internal/task/allocation.go

```diff
@@ -609,8 +607,10 @@ func (a *allocation) resourcesAllocated(msg *sproto.ResourcesAllocated) error {
 					switch {
 					case r.Rank == 0 && r.Started != nil && r.Started.Addresses != nil:
 						a.registerProxies(r.Started.Addresses)
+						a.closers = append(a.closers, a.unregisterProxies)
 					case a.model.ProxyAddress != nil:
 						a.registerProxies(a.containerProxyAddresses())
+						a.closers = append(a.closers, a.unregisterProxies)
```

Comment:
can we just always call this? ```unregisterProxies``` seems to have a len(0) check

Its weird to me that we call this 4 times. Maybe its more of a symptom of our proxy handling should be less complicated we have so many places we need to call this.